lightly/models/spark.py

```diff
@@ -0,0 +1,280 @@
+# Code adapted from https://github.com/keyu-tian/SparK/blob/main/
+
+import math
+from typing import List, Literal, Tuple, Union
+from pprint import pformat
+import sys
+
+import torch
+import torch.nn as nn
+from lightly.models.sparse.encoder import SparseResnet
+from lightly.models.modules.sparse.spark import SparseBatchNorm2d, SparseSyncBatchNorm2d, SparseConvNeXtLayerNorm
+
+def is_pow2n(n: int) -> bool:
+    return n > 0 and (n & (n - 1)) == 0
+
+
+class UNetBlock(nn.Module):
+    def __init__(self, cin, cout, bn2d):
+        """
+        a UNet block with 2x up sampling
+        """
+        super().__init__()
+        self.up_sample = nn.ConvTranspose2d(
+            cin, cin, kernel_size=4, stride=2, padding=1, bias=True
+        )
+        self.conv = nn.Sequential(
+            nn.Conv2d(cin, cin, kernel_size=3, stride=1, padding=1, bias=False),
+            bn2d(cin),
+            nn.ReLU6(inplace=True),
+            nn.Conv2d(cin, cout, kernel_size=3, stride=1, padding=1, bias=False),
+            bn2d(cout),
+        )
+
+    def forward(self, x):
+        x = self.up_sample(x)
+        return self.conv(x)
+
+
+class LightDecoder(nn.Module):
+    def __init__(self, up_sample_ratio, width=768, sync_batch_norm=True):
+        super().__init__()
```

Comment:
```suggestion
    def __init__(self, up_sample_ratio, width=768):
```

It shouldn't be necessary to add `sync_batch_norm` as argument as this can be handled outside the class. Torch has utility functions to automatically replace batch norms with sync batch norm.